src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -29,152 +30,196 @@
+
+		List<Conversation> conversations = new ArrayList<>();
+
+		// Retrieve all conversations from the datastore.
+		Query query = new Query("chat-conversations");
+		PreparedQuery results = datastore.prepare(query);
+
+		for (Entity entity : results.asIterable()) {
+			try {
+				UUID uuid = UUID.fromString((String) entity.getProperty("uuid"));
+				UUID ownerUuid = UUID.fromString((String) entity.getProperty("owner_uuid"));
+				String title = (String) entity.getProperty("title");
+				Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
+				boolean muted = Boolean.parseBoolean((String) entity.getProperty("muted"));
+				Conversation conversation = new Conversation(uuid, ownerUuid, title, creationTime, muted);
+				conversations.add(conversation);
+			} catch (Exception e) {
+				// In a production environment, errors should be very rare.
+				// Errors which may
+				// occur include network errors, Datastore service errors,
+				// authorization errors,
+				// database entity definition mismatches, or service mismatches.
+				throw new PersistentDataStoreException(e);
+			}
+		}
+
+		return conversations;
+	}
+
+	/**
+	 * Loads all Message objects from the Datastore service and returns them in
+	 * a List.
+	 *
+	 * @throws PersistentDataStoreException
+	 *             if an error was detected during the load from the Datastore
+	 *             service
+	 */
+	public List<Message> loadMessages() throws PersistentDataStoreException {
+
+		List<Message> messages = new ArrayList<>();
+
+		// Retrieve all messages from the datastore.
+		Query query = new Query("chat-messages");
+		PreparedQuery results = datastore.prepare(query);
+
+		for (Entity entity : results.asIterable()) {
+			try {
+				UUID uuid = UUID.fromString((String) entity.getProperty("uuid"));
+				UUID conversationUuid = UUID.fromString((String) entity.getProperty("conv_uuid"));
+				UUID authorUuid = UUID.fromString((String) entity.getProperty("author_uuid"));
+				Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
+				String content = (String) entity.getProperty("content");
+				Message message = new Message(uuid, conversationUuid, authorUuid, content, creationTime);
+				messages.add(message);
+			} catch (Exception e) {
+				// In a production environment, errors should be very rare.
+				// Errors which may
+				// occur include network errors, Datastore service errors,
+				// authorization errors,
+				// database entity definition mismatches, or service mismatches.
+				throw new PersistentDataStoreException(e);
+			}
+		}
+
+		return messages;
+	}
+
+	/**
+	 * Loads all Mention objects from the Datastore service and returns them in
+	 * a List.
+	 *
+	 * @throws PersistentDataStoreException
+	 *             if an error was detected during the load from the Datastore
+	 *             service
+	 */
+	public List<Mention> loadMentions() {
+		// TODO Auto-generated method stub
+		return null;
```

Comment:
I think you've got the hang of this!  Just use the same basic structure as loadUsers and it will work out fine (don't forget to add the throws PersistentDataStoreException clause).